app/pages/sections/order-items/catalogue-solutions/delete/routes.js

```diff
@@ -1,29 +1,69 @@
+  router.get('/:orderItemId/confirmation/:solutionName', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
+    const { orderId, orderItemId, solutionName } = req.params;
+    const accessToken = extractAccessToken({ req, tokenType: 'access' });
 
-    const context = {};
+    const context = await getDeleteCatalogueSolutionContext({
+      req,
+      sessionManager,
+      accessToken,
+      logger,
+    });
 
     logger.info(`navigating to order ${orderId} catalogue-solutions ${orderItemId} deletion page`);
-    return res.render('pages/sections/order-items/catalogue-solutions/dashboard/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
+    return res.render('pages/sections/order-items/catalogue-solutions/delete/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
   }));
 
-  router.post('/:orderItemId', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
-    const { orderId, orderItemId } = req.params;
+  router.post('/:orderItemId/confirmation/:solutionName', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
+    const { orderId, orderItemId, solutionName } = req.params;
     const accessToken = extractAccessToken({ req, tokenType: 'access' });
 
     await deleteCatalogueSolution({ orderId, orderItemId, accessToken });
 
     logger.info(`navigating to order ${orderId} catalogue-solution ${orderItemId} delete confirmation page`);
-    return res.redirect(`${config.baseUrl}/organisation/${orderId}/catalogue-solutions/delete/${orderItemId}/confirmation`);
+    return res.redirect(`${config.baseUrl}/organisation/${orderId}/catalogue-solutions/delete/${orderItemId}/confirmation/${solutionName}/continue`);
+  }));
+
+  router.get('/:orderItemId/confirmation/:solutionName/continue', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
+    const { orderId,  orderItemId, solutionName} = req.params;
+    const accessToken = extractAccessToken({ req, tokenType: 'access' });
+
+    const context = await getDeleteCatalogueSolutionConfirmationContext({
+      req,
+      sessionManager,
+      accessToken,
+      logger,
+    });
+
+    logger.info(`navigating to catalogue ${solutionName} delete-catalogue-confirmation page`);
+    res.render('pages/sections/order-items/catalogue-solutions/delete/confirmation/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
+  }));
+
+  router.post('/:orderItemId/confirmation/:solutionName/continue', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
+    const { orderId } = req.params;
+    
+    return res.redirect(`${config.baseUrl}/organisation/${orderId}/catalogue-solutions`);
```

Comment:
There appears to be a number of linting issues in this file:

```shell
app/pages/sections/order-items/catalogue-solutions/delete/routes.js
   3:51  error  'app/pages/sections/order-items/catalogue-solutions/delete/controller.js' imported multiple times  import/no-duplicates
   6:41  error  'app/pages/sections/order-items/catalogue-solutions/delete/controller.js' imported multiple times  import/no-duplicates
   9:3   error  'formatFormData' is defined but never used                                                                             no-unused-vars
  10:3   error  'getOrderItemContext' is defined but never used                                                                        no-unused-vars
  11:3   error  'getOrderItemErrorContext' is defined but never used                                                                   no-unused-vars
  12:3   error  'getPageData' is defined but never used                                                                                no-unused-vars
  13:3   error  'setEstimationPeriod' is defined but never used                                                                        no-unused-vars
  15:10  error  'validateOrderItemFormBulk' is defined but never used                                                                  no-unused-vars
  16:10  error  'saveOrderItemBulk' is defined but never used                                                                          no-unused-vars
  17:10  error  'sessionKeys' is defined but never used                                                                                no-unused-vars
  18:10  error  'transformApiValidationResponse' is defined but never used                                                             no-unused-vars
  24:35  error  'solutionName' is assigned a value but never used                                                                      no-unused-vars
  49:13  error  'orderId' is assigned a value but never used                                                                           no-unused-vars
  49:21  error  Multiple spaces found before 'orderItemId'                                                                             no-multi-spaces
  49:23  error  'orderItemId' is assigned a value but never used                                                                       no-unused-vars
  49:48  error  A space is required before '}'                                                                                         object-curly-spacing
  65:1   error  Trailing spaces not allowed                                                                                            no-trailing-spaces
```

Reply:
resolved